tagstore/models/taxonomy.py

```diff
@@ -1,16 +1,68 @@
 """
 A taxonomy is a collection of tags that can be applied to content.
 """
-from typing import NamedTuple, Optional, NewType
+from typing import Any, Iterator, NamedTuple, NewType, Optional, Tuple
+
+from .tag import Tag
 from .user import UserId
 
 TaxonomyId = NewType('TaxonomyId', int)
 
 
-class TaxonomyMetadata(NamedTuple):
+class Taxonomy(NamedTuple):
     """
     A taxonomy is a collection of tags that can be applied to content.
+
+    Is a NamedTuple for performance, simplicity, and immutability, but
+    we can change it to a full class at some point if needed.
     """
     uid: TaxonomyId
     name: str
     owner_id: Optional[UserId]
+
+    tagstore: Any  # Type is Tagstore but we can't define that without circular import
+
+    # Convenience methods:
+
+    def add_tag(self, name: str, parent_tag: Optional[Tag] = None) -> Tag:
+        """
+        Add the specified tag to this given taxonomy, and retuns it.
+
+        Will be a no-op if the tag already exists in the taxonomy (case-insensitive),
+        however the returned (existing) Tag may differ in case.
```

Comment:
I wasn't able to find a case where the "returned (existing) Tag may differ in case" occurrs..  suggest something like this instead?

> If a Tag already exists in the taxonomy with the given name (case-insensitive) with the given parent, then that Tag is returned and no changes are made.

Reply:
> I wasn't able to find a case where the "returned (existing) Tag may differ in case" 

There is a test case showing that? https://github.com/open-craft/blockstore/blob/05a88e283d81f67bb7094df6d7a453ef40ab3890/tagstore/backends/tests.py#L73-L75

I do like your wording though, so I'll use it.

Reply:
Looks like your change to the wording hasn't make it in yet?

Reply:
It did make it in in `tagstore.py` but I missed `taxonomy.py`. Fixed now :)